realsense_camera/src/base_nodelet.cpp

```diff
@@ -1066,6 +1087,82 @@ namespace realsense_camera
   }
 
   /*
+   * Publish Dynamic transforms.
+   */
+  void BaseNodelet::publishDynamicTransforms()
+  {
+    tf::Transform tr;
+    tf::Quaternion q;
+
+    // The color frame is used as the base frame.
+    // Hence no additional transformation is done from base frame to color frame.
+    tr.setOrigin(tf::Vector3(0,0,0));
+    tr.setRotation(tf::Quaternion(0, 0, 0, 1));
+    dynamic_tf_broadcaster_.sendTransform(tf::StampedTransform(tr, transform_ts_,
+          base_frame_id_, frame_id_[RS_STREAM_COLOR]));
+
+    // Transform color frame to color optical frame
+    tr.setOrigin(tf::Vector3(0,0,0));
+    q.setEuler(M_PI/2, 0.0, -M_PI/2);
+    tr.setRotation(q);
+    dynamic_tf_broadcaster_.sendTransform(tf::StampedTransform(tr, transform_ts_,
+          frame_id_[RS_STREAM_COLOR], optical_frame_id_[RS_STREAM_COLOR]));
+
+    // Transform base frame to depth frame
+    tr.setOrigin(tf::Vector3(
+           color2depth_extrinsic_.translation[2],
+          -color2depth_extrinsic_.translation[0],
+          -color2depth_extrinsic_.translation[1]));
+    tr.setRotation(tf::Quaternion(0, 0, 0, 1));
+    dynamic_tf_broadcaster_.sendTransform(tf::StampedTransform(tr, transform_ts_,
+          base_frame_id_, frame_id_[RS_STREAM_DEPTH]));
+
+    // Transform depth frame to depth optical frame
+    tr.setOrigin(tf::Vector3(0,0,0));
+    q.setEuler(M_PI/2, 0.0, -M_PI/2);
+    tr.setRotation(q);
+    dynamic_tf_broadcaster_.sendTransform(tf::StampedTransform(tr, transform_ts_,
+          frame_id_[RS_STREAM_DEPTH], optical_frame_id_[RS_STREAM_DEPTH]));
+
+    // Transform base frame to infrared frame
+    tr.setOrigin(tf::Vector3(
+           color2ir_extrinsic_.translation[2],
+          -color2ir_extrinsic_.translation[0],
+          -color2ir_extrinsic_.translation[1]));
+    tr.setRotation(tf::Quaternion(0, 0, 0, 1));
+    dynamic_tf_broadcaster_.sendTransform(tf::StampedTransform(tr, transform_ts_,
+          base_frame_id_, frame_id_[RS_STREAM_INFRARED]));
+
+    // Transform infrared frame to infrared optical frame
+    tr.setOrigin(tf::Vector3(0,0,0));
+    q.setEuler(M_PI/2, 0.0, -M_PI/2);
+    tr.setRotation(q);
+    dynamic_tf_broadcaster_.sendTransform(tf::StampedTransform(tr, transform_ts_,
+          frame_id_[RS_STREAM_INFRARED], optical_frame_id_[RS_STREAM_INFRARED]));
+  }
+
+  /*
+   * Prepare transforms.
+   */
+  void BaseNodelet::prepareTransforms()
+  {
+    // Publish transforms for the cameras
+    ROS_INFO_STREAM(nodelet_name_ << " - Publishing camera transforms (/tf)");
+
+    ros::Rate loop_rate(1);
```

Comment:
Any reason for such a slow loop? tf transforms are traditionally published at ~50Hz. Using here 1Hz causes a lot of `OLD_TF_DATA` errors when replaying a recorded bag of videos...